### ISSUE TITLE
Visualizations missing and/or broken

### ISSUE THREAD

**Issue by tekstryder** [clementine-player/Clementine#7151]
When attempting to view visualizations, only a single bouncing 'M' is available. The dropdown list is empty.

Noticed this on two separate Arch machines. Cannot find any modern issue-tracker item for this, and random forum posts could not provide any useful information.

Clementine version: 
1.4.0rc1+776+gcefe81d0c

projectm version 3.1.12

OS: Arch Linux x86_64    
Kernel: 5.15.21-1-lts    
Shell: bash 5.1.16    
DE: GNOME 41.3    
WM: Mutter    
Terminal: gnome-terminal    
CPU: 12th Gen Intel i9-12900K (24) @ 6.500GHz    
GPU: Intel AlderLake-S GT1    
GPU: NVIDIA GeForce GTX 1050 Ti    
Memory: 6456MiB / 64155MiB

**Comment by Lalarian** [comment]
Yep, same problem here. :(

**Comment by polsvoice** [comment]
I'm on Manjaro using 1.4 rc2, and I have the same problem. What I did in the past was compile it myself, but this time I can't do that because I keep getting
```
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles". 
 CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```
So, I installed Clementine from the repositories. ProjectM was installed automatically with it. I can see the presets in `/usr/share/projectM/presets`, but I can't figure out how to tell Clementine where they are.

**Comment by polsvoice** [comment]
So, I successfully compiled Clementine, and I found out that it stores visualizations in `/usr/local/share/clementine/projectm-presets/`. And those work. When you install Clementine from the repos, though, ProjectM stores its visualizations in `/usr/share/projectM/presets`. Since the set of visualizations I got from compiling was a lot smaller than all of the ones I got from installing ProjectM, I tried copying the other presets over to see if they'd show up in the visualizations menu. And they do!

So, if you copy the presets from the folders in `/usr/share/projectM/presets` to `/usr/local/share/clementine/projectm-presets/`, that should work. I didn't copy the folders themselves, because I wasn't sure that Clementine would search recursively. I just copied the contents (all of the `*.milk` files).

**Comment by kblaschke** [comment]
After we've finished the remaining work on the projectM 4.0 release, I'll see what I can do to improve the projectM visualization integration via a pull request.

Clementine currently uses an old copy of projectM's sources with a few custom patches to make it work due to some (alrady fixed) bugs in projectM, so these will no longer be required in the next release and Clementine can just use the shared library. I'll try to improve the settings page so users can easily add their own preset path(s) (and possibly also preset playlists).

The new release will also contain a ton of rendering fixes, making the visuals look more like Milkdrop's output (still not perfect though).

If you're interested in the development progress, you can follow our work here:
https://github.com/projectM-visualizer/projectm/projects/2

**Comment by Szewcson** [comment]
Still broken for debian, and repo version had visualizations disabled.

**Comment by kblaschke** [comment]
Yeah, I don't really expect something to happen anymore here, Clementine's been dead for a couple of years now. Strawberry is probably the way to go, but they've also remove visualization support entirely.

**Comment by Szewcson** [comment]
Development is still ongoing here from what I see. The last unstable release was 5 days ago...

**Comment by DarthGandalf** [comment]
#7306 fixes it

**Comment by tekstryder** [comment]
> #7306 fixes it

Cool! When that merge lands in a release I'll reinstall Clementine and check it out.
